File: src/052. Permuted multiples/052.rb
class Integer
  def to_sorted
    self.to_s.chars.sort.join
  end
end

n = 1
while true
  sorted = n.to_sorted
  if (n * 2).to_sorted == sorted and
     (n * 3).to_sorted == sorted and
     (n * 4).to_sorted == sorted and
     (n * 5).to_sorted == sorted and
     (n * 6).to_sorted == sorted
    puts n
    break
  end
  n += 1
end

File: src/052. Permuted multiples/052.java
import java.util.Arrays;

public class Main {

    public static void main(String[] args) {
        long n = 1;
        while (true) {
            String sorted = toSorted(n);
            if (sorted.equals(toSorted(n * 2))
                && sorted.equals(toSorted(n * 3))
                && sorted.equals(toSorted(n * 4))
                && sorted.equals(toSorted(n * 5))
                && sorted.equals(toSorted(n * 6))) {
                System.out.println(n);
                break;
            }
            n++;
        }
    }

    private static String toSorted(long n) {
        return sortString(Long.toString(n));
    }

    private static String sortString(String original) {
        char[] chars = original.toCharArray();
        Arrays.sort(chars);
        return new String(chars);
    }
}
